contracts/implementations/MultipleMinter.sol

```diff
@@ -5,51 +5,51 @@ import "./nodes/VehicleInternal.sol";
 import "./charging/ChargingInternal.sol";
 import "../interfaces/INFT.sol";
+import "../interfaces/ISacd.sol";
 import "../Eip712/Eip712CheckerInternal.sol";
 import "../libraries/NodesStorage.sol";
 import "../libraries/nodes/ManufacturerStorage.sol";
 import "../libraries/nodes/VehicleStorage.sol";
 import "../libraries/nodes/SyntheticDeviceStorage.sol";
 import "../libraries/MapperStorage.sol";
 import "../libraries/SharedStorage.sol";
+import "../shared/Errors.sol" as Errors;
 
 import {MINT_VEHICLE_OPERATION} from "../shared/Operations.sol";
-import "../shared/Roles.sol";
-
-import "@solidstate/contracts/access/access_control/AccessControlInternal.sol";
 
 error DeviceAlreadyRegistered(address addr);
 error InvalidSdSignature();
 
-contract MultipleMinter is
-    AccessControlInternal,
-    VehicleInternal,
-    SyntheticDeviceInternal
-{
+contract MultipleMinter is VehicleInternal, SyntheticDeviceInternal {
     bytes32 private constant MINT_VEHICLE_SD_TYPEHASH =
         keccak256("MintVehicleAndSdSign(uint256 connectionId)");
 
     /**
      * @notice Mints and pairs a vehicle and a synthetic device through a metatransaction
-     * The vehicle owner signs a typed structured (EIP-712) message in advance and submits to be verified//-
-     * @dev Caller must have the MINT_VEHICLE_SD_ROLE
+     * @dev The vehicle owner signs a typed structured (EIP-712) message in advance which is verified during execution.
+     * Caller must have the CONNECTION_MINT_SD_PERMISSION for the specified connection or be the connection ID owner
      * @param data A MintVehicleAndSdInput struct containing:
-     *        - manufacturerNode: Parent manufacturer node id of the vehicle
-     *        - owner: The new nodes owner
+     *        - manufacturerNode: Parent manufacturer node ID of the vehicle
+     *        - owner: The new nodes owner address
      *        - attrInfoPairsVehicle: List of attribute-info pairs to be added to the vehicle
-     *        - connectionId: Parent connection id of the synthetic device
-     *        - vehicleOwnerSig: Vehicle owner signature hash
-     *        - syntheticDeviceSig: Synthetic Device's signature hash
+     *        - connectionId: Parent connection ID of the synthetic device
+     *        - vehicleOwnerSig: Vehicle owner's EIP-712 signature
+     *        - syntheticDeviceSig: Synthetic device's EIP-712 signature
      *        - syntheticDeviceAddr: Address associated with the synthetic device
      *        - attrInfoPairsDevice: List of attribute-info pairs to be added to the synthetic device
+     * Emits VehicleNodeMinted and SyntheticDeviceNodeMinted events on success.
+     * Reverts if connection or manufacturer node doesn't exist, caller lacks permission,
+     * device is already registered, or signatures are invalid.
      */
     function mintVehicleAndSdSign(
         MintVehicleAndSdInput calldata data
-    ) external onlyRole(MINT_VEHICLE_SD_ROLE) {
```

Comment:
By the way, is anyone calling this anymore? I think we even stopped listening for it. Can we get rid of it? Doesn't have to be in this PR.

Reply:
Keeping track here https://linear.app/dimo/issue/EX-3708/remove-unused-dimoregistry-functions
Maybe we find more